internal/handlers/common/update.go

```diff
@@ -26,16 +26,18 @@ import (
 // UpdateDocument updates the given document with a series of update operators.
 // Returns true if document was changed.
 func UpdateDocument(doc, update *types.Document) (bool, error) {
+	var changed bool
 	for _, updateOp := range update.Keys() {
 		updateV := must.NotFail(update.Get(updateOp))
 
 		switch updateOp {
-		case "$set":
+		case "$set",
+			"$setOnInsert":
```

Comment:
This formatting looks a bit unusual to me. But I'm not sure I know the right formatting :) Why are both conditions not on the same line? Is it something linter-specific?

Reply:
> This formatting looks a bit unusual to me. But I'm not sure I know the right formatting :) Why are both conditions not on the same line? Is it something linter-specific?

a new test case was added:
```go
                "set-setoninsert": {
			filter: bson.D{{"_id", "test"}},
			query: bson.D{
				{"$set", bson.D{{"foo", int32(12)}}},
				{"$setOnInsert", bson.D{{"value", math.NaN()}}},
			},
			res: bson.D{{"_id", "test"}, {"foo", int32(12)}, {"value", math.NaN()}},
		},
```
and it was found that when multiply changes in $set clause then `modified` count gives the wrong number.

Reply:
Linters are ok with that.
As for me it's simpler to read and simpler to see changes during the time.
Imagine you adding a new operator.
If these operators were in one line, you should parse the line with your eyes and see the change.
But if they are each in the new line, then GitHub shows you the change.

That's it.

Reply:
Let's not use that formatting. We have two existing options already, let's use one of them:

```go
case "$set", "$setOnInsert":
```

or

```go
case "$set":
  fallthrough
case "$setOnInsert":
```